fdbserver/DataDistribution.actor.cpp

```diff
@@ -1736,7 +1796,9 @@ struct DDTeamCollection : ReferenceCounted<DDTeamCollection> {
 			addedMachineTeams = addBestMachineTeams(machineTeamsToBuild, remainingMachineTeamBudget);
 		}
 
-		while (addedTeams < teamsToBuild || addedTeams < remainingTeamBudget) {
+		int minTeamNumPerServer = getMinTeamNumPerServer();
+		while (addedTeams < teamsToBuild || addedTeams < remainingTeamBudget ||
```

Comment:
minTeamNumPerServer and remainingTeamBudget are the same

Reply:
This turns out not the same. The remainingTeamBudget is the total amount of teams we are supposed to build. It is possible that we build teams that are not on the server with the second least number of servers. 
Then the addTeamsBestOf() will exit before it builds enough teams such that each server has desired_team_number * (storageTeamSize + 1) / 2 teams. 
This leads to failure of consistency check.